common/beerocks/bwl/dummy/mon_wlan_hal_dummy.cpp

```diff
@@ -80,6 +114,27 @@ bool mon_wlan_hal_dummy::sta_channel_load_11k_request(const SStaChannelLoadReque
 bool mon_wlan_hal_dummy::sta_beacon_11k_request(const SBeaconRequest11k &req, int &dialog_token)
 {
     LOG(TRACE) << __func__;
+
+    auto sta_mac = beerocks::net::network_utils::mac_to_string(req.sta_mac.oct);
+    auto bssid   = beerocks::net::network_utils::mac_to_string(req.bssid.oct);
+
+    LOG(DEBUG) << "Beacon 11k request to sta " << sta_mac << " on bssid " << bssid << " channel "
+               << std::to_string(req.channel);
+
+    auto dummy_sta = m_dummy_stas_map.find(sta_mac);
+    if (dummy_sta != m_dummy_stas_map.end()) {
+        auto dummy_bssid_event = dummy_sta->second.beacon_measurment_events.find(bssid);
+        if (dummy_bssid_event == dummy_sta->second.beacon_measurment_events.end()) {
+            LOG(WARNING) << "No RRM beacon response for bssid " << bssid;
+        } else {
+            //TODO: remove sleep. Sleep fixes missing requests sent by controller but not
+            //received by agent
+            sleep(1);
+            process_dummy_event(dummy_bssid_event->second);
```

Comment:
This is the only place where the RRM event is used. So instead, that code can be moved here directly.

Reply:
See comment above.